### PR TITLE
Add TxIn::pegin_prevout method to return Bitcoin prevout

### PR DESCRIPTION
**Pull request by stevenroose**
This kind of makes the case "safer" for users.

### CODE REVIEW

File: src/transaction.rs [hunk 1]
```diff
@@ -295,11 +308,7 @@ impl TxIn {
         );
 
         Some(PeginData {
-            // Cast of an elements::OutPoint to a bitcoin::OutPoint
-            outpoint: bitcoin::OutPoint {
-                txid: bitcoin::Txid::from(self.previous_output.txid.as_hash()),
-                vout: self.previous_output.vout,
-            },
+            outpoint: self.pegin_prevout().expect("this is a pegin input"),
```
Comment (sanket1729) [inline src/transaction.rs:311]: nit: You can also `?` here(since it is can already fail) to avoid future readers of the code thinking about this.

**Review by sanket1729 — APPROVED** [review]
ACK 1195a43bdaaf68eef4482d0dbec860aaf0e5c4b7. Really a nit, feel free to ignore.

File: src/transaction.rs (hunk 1 above)
Comment (apoelstra) [reply src/transaction.rs:311]: Could also change the top `if self.is_pegin` check to an `if let` on this data.

Comment (stevenroose) [reply src/transaction.rs:311]: addressed these nits

Comment (stevenroose) [reply src/transaction.rs:311]: ah damn had this locally fixed but didn't push that, my bad

**Review by sanket1729 — APPROVED** [review]
ACK e52a71fd0acf764883fe0fd31c11e43731523fda

**Review by apoelstra — APPROVED** [review]
ACK 2085ad4eeddc92f40bd20847c93fc6362b5c964f

**Pull request merged**